src/client.ts

```diff
@@ -358,20 +358,20 @@ export default class Client implements ClientInterface {
     }
 
     try {
-      // If bundle show succeeds, it means the ruby-lsp gem is a part of the bundle
+      // detect if ruby-lsp is a direct dependency, but ignore if it's transitive
       await asyncExec(
         `BUNDLE_GEMFILE=${path.join(
           this.workingFolder,
           "Gemfile"
-        )} bundle show ruby-lsp`,
+        )} bundle exec ruby -e "exit 1 if Bundler.locked_gems.dependencies.key?('ruby-lsp')"`,
         {
           cwd: this.workingFolder,
           env: this.ruby.env,
         }
       );
-      return true;
-    } catch (error) {
       return false;
+    } catch (error) {
+      return true;
```

Comment:
Something is confusing here, I'm seeing a breakpoint hit for `return true`, but then it hits the `else` clause in `if await this.rubyLspIsInBundle()`, am I missing some subtlety of the async behaviour @vinistock ?

Reply:
I'm not 100% sure. But my first suggestion would be to invert the logic to make the success case `true`. That is
```
exit 1 unless Bundler.locked_gems.dependencies.key?('ruby-lsp')
```

Did you also confirm that exiting with 1 makes `asyncExec` throw? I don't know what the conditions are for it throwing. Does it throw on any non zero exit status?

Reply:
> Does it throw on any non zero exit status?

Yes: https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback

Reply:
I'd still prefer us to invert the logic though. It feels confusing to me that the error is the true case.

Reply:
Hmmm, to me it feels more natural this way because having `ruby-lsp` in the Gemfile is what we **don't** want.

@st0012 help us break the tie? :grin: